Source/WebCore/css/parser/CSSParserFastPaths.cpp

```diff
@@ -162,25 +162,25 @@ static inline bool parseSimpleAngle(std::span<const CharacterType> characters, R
 }
 
 template <typename CharacterType>
-static inline bool parseSimpleNumberOrPercentage(std::span<const CharacterType> characters, ValueRange valueRange, CSSUnitType& unit, double& number)
+static inline bool parseSimpleNumberOrPercentageDividedBy100(std::span<const CharacterType> characters, double& number)
 {
-    unit = CSSUnitType::CSS_NUMBER;
+    bool isPercentage = false;
     if (!characters.empty() && characters.back() == '%') {
         dropLast(characters);
```

Comment:
Is it desirable to do this (without undoing it) even if we fail to parse a number and subsequently return `false`?

Reply:
Generally, I would say no, as I prefer these kind of consume functions to do nothing if they fail, it doesn't seem like that is followed in the fast path, so for now, I am think keeping it as is makes sense.